akka-testkit-typed/src/main/scala/akka/testkit/typed/scaladsl/ActorTestKit.scala

```diff
@@ -0,0 +1,130 @@
+package akka.testkit.typed.scaladsl
+
+import akka.actor.typed.scaladsl.AskPattern._
+import akka.actor.typed.{ ActorRef, ActorSystem, Behavior, Props }
+import akka.annotation.{ ApiMayChange, InternalApi }
+import akka.testkit.typed.TestKitSettings
+import akka.testkit.typed.internal.{ ActorTestKitGuardian, TestKitUtils }
+import com.typesafe.config.{ Config, ConfigFactory }
+
+import scala.concurrent.Await
+import scala.concurrent.duration._
+
+object ActorTestKit {
+
+  /**
+   * Shutdown the given [[akka.actor.typed.ActorSystem]] and block until it shuts down,
+   * if more time than `TestKitSettings.DefaultActorSystemShutdownTimeout` passes an exception is thrown
+   */
+  def shutdown(system: ActorSystem[_]): Unit = {
+    val settings = TestKitSettings(system)
+    TestKitUtils.shutdown(
+      system,
+      settings.DefaultActorSystemShutdownTimeout,
+      settings.ThrowOnShutdownTimeout
+    )
+  }
+
+  /**
+   * Shutdown the given [[akka.actor.typed.ActorSystem]] and block until it shuts down
+   * or the `duration` hits. If the timeout hits `verifySystemShutdown` decides
+   */
+  def shutdown(
+    system:               ActorSystem[_],
+    timeout:              Duration,
+    throwIfShutdownFails: Boolean        = false): Unit =
+    TestKitUtils.shutdown(system, timeout, throwIfShutdownFails)
+
+  // place holder for no custom config specified to avoid the boilerplate
+  // of an option for config in the trait
+  private val noConfigSet = ConfigFactory.parseString("")
+
```

Comment:
we could add factory methods for creating `ActorTestKit` here also (as alternative to extending), similar to the javadsl, but we can do that later if needed

Reply:
I think we should make people do it in one way rather than provide two almost the same ways out of the box. Let's add that if someone asks for it.

Reply:
I'd say that "use as a tool" is better than inheriting. E.g. you can more easily create multiple in same test. There is no risk of name clashes when we add more methods. We have started using that style in several of the testing tools, such as TestProbe, BehaviorTestkit.

So if I have to pick one style I'd choose that one.
On the other hand I have some understanding for ScalaTest style which loves mixing in stuff.

Reply:
Hmm, that's a fair point, I think I leaned this direction just because of the old testkit design. Maybe we should default like that, just like we do for the Java API then.

Reply:
Let's merge this and follow up on this topic separately

Reply:
Made a separate issue to track it #24598